### PR TITLE
Test pixelMove UFE support.

### PR DESCRIPTION
**Pull request by ppt-adsk**
This change tests the new UFE support in the pixelMove command.  This addresses
https://github.com/Autodesk/maya-usd/issues/2234

### CODE REVIEW

File: test/testUtils/mayaUtils.py [hunk 1]
```diff
@@ -298,6 +298,16 @@ def mayaMajorMinorVersions():
     """
     return (mayaMajorVersion(), mayaMinorVersion())
 
+def ufeSupportFixLevel():
```
Comment (ppt-adsk) [inline test/testUtils/mayaUtils.py:301]: Return the newly-implemented UFE fix level, to determine if the UFE pixelMove support is present in Maya or not.  fixLevel is implemented in Maya, in the ufeSupport.utils module.  If the fixLevel function does not exist, returns 0.

File: test/lib/ufe/testPixelMoveCmd.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+from testUtils import assertVectorAlmostEqual, assertVectorNotAlmostEqual
+
+import mayaUsd.lib
+
+from maya import cmds
+from maya import standalone
+
+import ufe
+
+@unittest.skipUnless(mayaUtils.ufeSupportFixLevel() > 0, "Requires pixelMove UFE support.")
```
Comment (ppt-adsk) [inline test/lib/ufe/testPixelMoveCmd.py:32]: UFE support for pixelMove is implemented at fix level 1, so don't run if fix level is below that.

File: test/lib/ufe/testPixelMoveCmd.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+    @classmethod
+    def setUpClass(cls):
+        fixturesUtils.readOnlySetUpClass(__file__, initializeStandalone=False)
+
+    def setUp(self):
+        cmds.file(new=True, force=True)
+        cmds.select(clear=True)
+    
+    def testPixelMove(self):
```
Comment (ppt-adsk) [inline test/lib/ufe/testPixelMoveCmd.py:43]: Test UFE pixelMove support by comparing with a Maya object.  Move for Maya object and move for non-Maya object (i.e. USD) must match.

File: test/lib/ufe/CMakeLists.txt [hunk 4]
```diff
@@ -43,6 +43,7 @@ if(CMAKE_UFE_V2_FEATURES_AVAILABLE)
     )
 
     list(APPEND INTERACTIVE_TEST_SCRIPT_FILES
+        testPixelMoveCmd.py
```
Comment (ppt-adsk) [inline test/lib/ufe/CMakeLists.txt:46]: pixelMove requires a view and a camera, so must be an interactive test.

**Comment by ppt-adsk** [comment]
Only failure is master OSX Python 3 only, with clang compiler crash while compiling mayaShaderTranslation.cpp, appears to be a random failure.

File: test/lib/ufe/testPixelMoveCmd.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        '''Pixel move command must move non-Maya UFE objects.'''
+
+        # pixelMove has different behavior for orthographic and perspective
+        # views.  Only perspective views are affected by the selected objects'
```
Comment (williamkrick) [inline test/lib/ufe/testPixelMoveCmd.py:47]: Should we also test doing pixel move while looking through an ortho camera?

**Review by williamkrick — APPROVED** [review]

File: test/lib/ufe/testPixelMoveCmd.py (hunk 5 above)
Comment (ppt-adsk) [reply test/lib/ufe/testPixelMoveCmd.py:47]: Ortho camera pixel move is unaffected by the selection, and therefore did not require UFE-related changes.  Testing pixel move with ortho camera is essentially testing Maya behavior, which plugins should simply rely on and expect (hope?) is tested in Maya.

**Pull request merged**